launchable/commands/inspect/tests.py

```diff
@@ -1,23 +1,154 @@
+import json
 import os
 import sys
+from abc import ABCMeta, abstractmethod
 from http import HTTPStatus
+from typing import List
 
 import click
 from tabulate import tabulate
 
 from ...utils.env_keys import REPORT_ERROR_KEY
 from ...utils.launchable_client import LaunchableClient
+from ...utils.session import parse_session
+from ..helper import require_session
+
+
+class TestResult(object):
+    def __init__(self, result: dict):
+        self._status = result.get("status", "")
+        self._duration_sec = result.get("duration", 0.0)
+        self._created_at = result.get("createdAt", None)
+        self._test_path = "#".join([path["type"] + "=" + path["name"]
+                                   for path in result["testPath"] if path.keys() >= {"type", "name"}])
+
+
+class TestResults(object):
+    def __init__(self, results: List[TestResult]):
+        self._results = results
+
+    def add(self, result: TestResult):
+        self._results.append(result)
+
+    def list(self) -> List[TestResult]:
+        return self._results
+
+    def total_duration_sec(self) -> float:
+        return sum([result._duration_sec for result in self._results])
+
+    def total_duration_min(self) -> float:
+        return (sum([result._duration_sec for result in self._results]) / 60)
+
+    def total_count(self) -> int:
+        return len(self._results)
+
+    def filter_by_status(self, status: str) -> 'TestResults':
```

Comment:
In the future, let's use Self Type.
https://peps.python.org/pep-0673/

Reply:
Right, we need to support python 3.7 now